spark/v3.5/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestViews.java

```diff
@@ -562,6 +562,64 @@ public void readFromViewWithCTE() throws NoSuchTableException {
     assertThat(sql("SELECT * FROM %s", viewName)).hasSize(1).containsExactly(row(10, 1L));
   }
 
+  @TestTemplate
+  public void readFromViewWithGroupByOrdinal() throws NoSuchTableException {
+    insertRows(3);
+    insertRows(2);
+    String viewName = viewName("viewWithGroupByOrdinal");
+    String sql = String.format("SELECT id, count(1) FROM %s GROUP BY 1", tableName);
+
+    ViewCatalog viewCatalog = viewCatalog();
+
+    viewCatalog
+        .buildView(TableIdentifier.of(NAMESPACE, viewName))
+        .withQuery("spark", sql)
+        .withDefaultNamespace(NAMESPACE)
+        .withDefaultCatalog(catalogName)
+        .withSchema(schema(sql))
+        .create();
+
+    assertThat(sql("SELECT * FROM %s", viewName))
+        .hasSize(3)
+        .containsExactlyInAnyOrder(row(1, 2L), row(2, 2L), row(3, 1L));
+  }
+
+  @TestTemplate
+  public void readFromViewWithOrderByOrdinal() throws NoSuchTableException {
+    insertRows(10);
+    String viewName = viewName("viewWithOrderByOrdinal");
+    String sql = String.format("SELECT id FROM %s ORDER BY 1", tableName);
```

Comment:
this test doesn't seem to fail without the fix

Reply:
yes，I checked that it is indeed the case. deleted

Reply:
is there a way you could actually make this particular test fail without the fix and pass with the fix?

Reply:
I have no problem debugging "order by ordinals" case .